src/dune_rules/modules.mli

```diff
@@ -92,7 +92,7 @@ module Sourced_module : sig
     | Impl_of_virtual_module of Module.t Ml_kind.Dict.t
 end
 
-val obj_map : t -> f:(Sourced_module.t -> 'a) -> 'a Module.Obj_map.t
+val obj_map : t -> Module.t Module_name.Unique.Map.t
```

Comment:
Should we just make this valued by `Sourced_module.t`? It doesn't really hurt for `ocamldep`, and it will remove further redundant computations.

Reply:
Do you mean the type being 

```
val obj_map : t -> Sourced_module.t Module_name.Unique.Map.t
```
?
I tried something like that in https://github.com/ocaml/dune/pull/7187/commits/c8e81be6742744b558079a4fafc5a0d1ba6b3e59, but then I get errors in `deps_of_module` and `deps_of_vlib_module` because of the change, as some functions like `Modules.alias_for` still operate on `Module.t`.

Reply:
```
File "src/dune_rules/dep_rules.ml", line 62, characters 16-20:
62 |       aliases @ deps)
                     ^^^^
Error: This expression has type Modules.Sourced_module.t list
       but an expression was expected of type Module.t list
       Type Modules.Sourced_module.t is not compatible with type Module.t
```

Reply:
Right, but you can extract the `Module.t` out of the `Sourced_module.t` without too many issues.

Reply:
This builds, but I'm not sure I got the idea right, as it seems more complex than before 🤔 https://github.com/ocaml/dune/pull/7187/commits/559e885f0af7f1d9b4bc1b590496e8a637195de2

Reply:
I think you can simplify it by keeping `ocamldep.mli` the same, and moving the `to_module` conversion to `ocamldep.ml`.

Reply:
Tried something in https://github.com/ocaml/dune/pull/7187/commits/81d333991e480a7305a601fcfe6831146d06ebb1.